Sources/pipeablejs/src/pipeable.ts

```diff
@@ -271,26 +278,41 @@
             return false;
         }
 
-        focus(elementHash: string) {
-            const el = this.elementRegistry.get(elementHash);
-
+        _focus(el?: HTMLElement | Element) {
+            // Ref:
+            // * https://developer.mozilla.org/en-US/docs/Web/API/Element/focus_event
+            // * https://api.jquery.com/focus/
             if (el) {
                 // emit focus event.
                 const event = new FocusEvent('focus', {
                     view: window,
-                    bubbles: true,
-                    cancelable: true,
+                    bubbles: false,
+                    cancelable: false,
                 });
 
                 el.dispatchEvent(event);
-                el.focus();
+                if (el instanceof HTMLInputElement) {
+                    el.focus();
```

Comment:
Would this result in a generated `focusin` event? Or do we still need to synthesize it?

Reply:
Ah yes, good catch, `el.focus` will fire a `focus` event and a `focusin` event, so we are duplicating both, seems like we should try `el.focus()` first and if it isn't available then fall back to manually firing the `focus` and `focusin` events, what do you think?